client/test/webview/TutorialGalleryView.test.ts

```diff
@@ -0,0 +1,153 @@
+            reveal: (): void => {
+                return;
+            },
+            title: 'Tutorial Gallery',
+            onDidDispose: mySandBox.stub(),
+            onDidChangeViewState: mySandBox.stub()
+        });
+
+        View['openPanels'].splice(0, View['openPanels'].length);
+    });
+
+    afterEach(() => {
+        mySandBox.restore();
+    });
+
+    it('should register and show tutorial gallery page', async () => {
+
+        const sampleView: TutorialGalleryView = new TutorialGalleryView(context);
+        await sampleView.openView(false);
+        createWebviewPanelStub.should.have.been.called;
+    });
+
+    it('should reveal tutorial gallery page if already open', async () => {
+
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+        await tutorialGalleryView.openView(true);
+        await tutorialGalleryView.openView(true);
+
+        createWebviewPanelStub.should.have.been.calledOnce;
+
+        should.equal(createWebviewPanelStub.getCall(1), null);
+    });
+
+    it('should dispose tutorial gallery page', async () => {
+        const filterSpy: sinon.SinonSpy = mySandBox.spy(Array.prototype, 'filter');
+
+        createWebviewPanelStub.returns({
+            webview: {
+                onDidReceiveMessage: mySandBox.stub()
+            },
+            reveal: (): void => {
+                return;
+            },
+            title: 'Tutorial Gallery',
+            onDidDispose: mySandBox.stub().yields(),
+            onDidChangeViewState: mySandBox.stub()
+        });
+
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+
+        mySandBox.stub(tutorialGalleryView, 'getHTMLString').resolves('<html>Tutorial Gallery</html>');
+        await tutorialGalleryView.openView(true);
+
+        createWebviewPanelStub.should.have.been.calledWith(
+            'tutorialGallery',
+            'Tutorial Gallery',
+            vscode.ViewColumn.One,
+            {
+                enableScripts: true,
+                retainContextWhenHidden: true,
+                enableCommandUris: true,
+                localResourceRoots: [
+                    vscode.Uri.file(path.join(context.extensionPath, 'resources'))
+                ]
+            }
+        );
+
+        createWebviewPanelStub.should.have.been.calledOnce;
+
+        filterSpy.getCall(1).thisValue[filterSpy.getCall(1).thisValue.length - 1].webview.html.should.equal('<html>Tutorial Gallery</html>');
+    });
+
+    it('should do nothing on openPanelInner', async () => {
+        const tutorialGalleryView: TutorialGalleryView = new TutorialGalleryView(context);
+        await tutorialGalleryView['openPanelInner']();
+    });
+
+    it('should return string on getHTMLString', async () => {
+        const allSeries: any[] = [
+            {
+                some: 'series'
+            }
+        ];
+        mySandBox.stub(View.prototype, 'getAllSeries').resolves(allSeries);
```

Comment:
you don't need to stub this, it should get it from reading the json file

Reply:
I think stubbing this is a good idea as it's a unit test and whenever we update the `tutorials.json` we don't want to have to update the tests.

Reply:
yeah but then you would never notice if you broke something as we don't do integration tests for this and can't really do an integration test